### DISCUSSION TITLE
Additional parameter while adding items in collection

### DISCUSSION THREAD

**Discussion by sangitgrg** [dotnet/runtime#65832] [Ideas]
While adding new items in collections like in list we check if that item is present in list or not and then add 
ex :  _assuming listOfEmployee is List<string> with some values in it_

```
if (!listOfEmployee.Contains(employeeName))
    {
         listOfEmployee.Add(employeeName);
    }
```
Instead of writing additional 'if' statement ,
we could add a new parameter in Add method that will tell to check for existence of items in collection.
Additional parameter could be enum with list of options.
And the code to add item in collection could be something like this :
```
   listOfEmployee.Add(employeeName, option.AddOnlyIfNotExist)
   listOfEmployee.Add(employeeName, option.ReplaceAndAdd)
   listOfEmployee.Add(employeeName, option.DonotAddIfExist)
```

sorry for terrible option name :-)

**Comment by YairHalberstadt** [comment]
This is an issue for dotnet/runtime repository

**Comment by TonyValenti** [comment]
Are you familiar with Extension Methods? You might consider making an extension method that allows you to do that.

**Comment by Clockwork-Muse** [comment]
... are you sure you don't want `HashSet` or `Dictionary`?
